pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ */
+
+package mutators
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/api/admission/v1beta1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+type VMsMutator struct {
+	ClusterConfig *virtconfig.ClusterConfig
+}
+
+func (mutator *VMsMutator) Mutate(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	if !webhooks.ValidateRequestResource(ar.Request.Resource, webhooks.VirtualMachineGroupVersionResource.Group, webhooks.VirtualMachineGroupVersionResource.Resource) {
```

Comment:
@fromanirh so finally got it. The issue is related to what you meant. Until the code change in k8s you pointed at, the mutating webhook was not supposed to return errors. So instead of validating here and returning the validation error, we should just return **without** applying any defaults (I guess just leaving the patch type and body empty, but we have to try out). The validation webhook will then reject it properly.

Reply:
hey, great catch. Thanks for your help. Updating the patch accordingly